docs: full manifest update (#340)
edgelesssys/marblerun: docs/docs/workflows/update-manifest.md

# Updating a manifest

When [defining the initial manifest](define-manifest.md), you have to choose whether it should be [partially or fully updatable](../features/manifest.md) by assigning appropriate [roles](define-manifest.md#roles) to eligible users.

## Package updates

> Updates play an important role to ensure your software stays secure. To avoid having to redeploy your application from scratch, MarbleRun allows uploading a separate "update manifest" that increases the minimum `SecurityVersion` of already deployed packages. After such an update is performed, an old version of a defined software package can't be loaded anymore under the current manifest.

Reviewer: ```suggestion
Updates play an important role in ensuring your software stays secure. To avoid redeploying your application from scratch, MarbleRun allows uploading a separate "update manifest" that increases the minimum `SecurityVersion` of already deployed packages. After such an update is performed, an old version of a defined software package can't be loaded anymore under the current manifest.
```

To deploy an update, your user needs to have a [role assigned that contains the `UpdateSecurityVersion` action](define-manifest.md#roles).

### Defining an update manifest

> The format of an update manifest follows the syntax of the original manifest, but it only contains packages with new `SecurityVersion` values.

Reviewer: ```suggestion
The update manifest format follows the original manifest's syntax, but it only contains packages with new `SecurityVersion` values.
```

}
```

Don't define other values except the `SecurityVersion` value for a package, as MarbleRun refuses to accept such an update manifest.

## Full update

<enterpriseBanner/>

Some deployment scenarios require more flexibility regarding changes to the manifest. To this end, MarbleRun also allows uploading a full manifest. User-defined secrets and secrets of type `symmetric-key` are retained if their definition doesn't change.

> To deploy a new manifest, your user needs to have a [role assigned that contains the `UpdateManifest` action](define-manifest.md#roles).

Reviewer: ```suggestion
To deploy a new manifest, your user must have a [role assigned that contains the `UpdateManifest` action](define-manifest.md#roles).
```
